docs/model-catalog.md

```diff
@@ -0,0 +1,87 @@
+---
+title: "Model catalog"
+slug: /model-catalog
+date: 2024-12-26
+keyword: model catalog
+license: "This software is licensed under the Apache License version 2."
+---
+
+## Introduction
+
+Model catalog is a metadata catalog that offers the unified interfaces to manage the metadata of
+machine learning models in a centralized way. It follows the typical Gravitino 3-level namespace
+(catalog, schema, and model) to manage the ML models metadata. In additional, it supports
+managing the versions for each model.
+
+The advantages of using model catalog are:
+
+* Centralized management of ML models with user defined namespaces. Users can better discover
+  and govern the models from sematic level, rather than managing the model files directly.
+* Version management for each model. Users can easily track the model versions and manage the
+  model lifecycle.
+
+The key concept of model management is to manage the path (URI) of the model. Instead of
+managing the model storage path physically and separately, model metadata defines the mapping
+relation between the model name and the storage path. In the meantime, with the support of
+extensible properties of model metadata, users can define the model metadata with more detailed information
+rather than just the storage path.
+
+* **Model**: The model is a metadata defined in the model catalog, to manage the ML models. Each
+  model can have many **Model Versions**, and each version can have its own properties. Models
+  can be retrieved by the name.
+* **Model Version**: The model version is a metadata defined in the model catalog, to manage each
+  version of the ML model. Each version has a unique version number, and can have its own
+  properties and storage path. Model version can be retrieved by the model name and version
+  number. Also, each version can have a list of aliases, which can also be used to retrieve.
```

Comment:
If we want to use the term "model version" to represent an object, i.e. a specific version of a model, we may coin a term "ModelVersion", formally.
We then use this term consistently across the docs and code base.
We don't use the regular phrase "model version" which make cause confusion.